module/zfs/spa.c

```diff
@@ -1314,6 +1314,10 @@ spa_activate(spa_t *spa, spa_mode_t mode)
 	    spa_error_entry_compare, sizeof (spa_error_entry_t),
 	    offsetof(spa_error_entry_t, se_avl));
 
+#if defined(_KERNEL)
```

Comment:
Isn't it simpler to ifdef KERNEL && APPLE to get rid of empty stubs? Or was it decided against that?

Reply:
Definitely an option, up to this point, we've gone with empty stubs on the platforms that do not use them - as the platform can easily chose when wanting to use it without changing shared code.

Reply:
Right.  Our preference is to use the empty stubs whenever possible to keep the shared code as readable as possible.

One thing we could also do here is add a userspace version of these hooks to drop the need to the `_KERNEL` wrapper entirely.  That said, I don't see a great existing spot for them so I'm okay with it as is.